Compiler error in generated validation

In generated file xsd/types.rs
```rust
// Type used to represent the numbers from 1 ,2 , 3,...
#[derive(Default, PartialEq, Debug, UtilsTupleIo, UtilsDefaultSerde)]
pub struct PositiveInteger (pub u32);

impl Validate for PositiveInteger {
    fn validate(&self) -> Result<(), String> { 
        if self.0 < "1".parse().unwrap() {
            return Err(format!("MinInclusive validation error: invalid value of 0! \nExpected: 0 >= 1.\nActual: 0 == {}", self.0));
        }
        Ok(())
    }
}
```
Compiler result:
```
error[E0283]: type annotations needed
   --> src/schema/types.rs:106:19
	|
106 |     	if self.0 < "1".parse().unwrap() {
	|               	^ cannot infer type for type `u32`
	|
	= note: cannot resolve `u32: std::cmp::PartialOrd<_>`

```

--- a/xsd-parser/src/generator/tuple_struct.rs
+++ b/xsd-parser/src/generator/tuple_struct.rs
@@ -49,7 +49,7 @@ pub trait TupleStructGenerator {
         let body = entity
             .facets
             .iter()
-            .map(|f| gen_facet_validation(&f.facet_type, "0"))
+            .map(|f| gen_facet_validation(&f.facet_type, "0", &self.get_type_name(entity, gen)))
             .fold(String::new(), |x, y| (x + &y));
         Cow::Owned(gen_validate_impl(
             self.get_name(entity, gen).as_str(),

--- a/xsd-parser/src/generator/validator.rs
+++ b/xsd-parser/src/generator/validator.rs
@@ -26,47 +26,58 @@ pub fn gen_validate_impl(name: &str, body: &str) -> String {
     }
 }
 
-pub fn gen_facet_validation(facet: &FacetType, name: &str) -> Cow<'static, str> {
+pub fn gen_facet_validation(facet: &FacetType, name: &str, typename: &str) -> Cow<'static, str> {
     match facet {
         FacetType::Enumeration(_) => "".into(),
         FacetType::FractionDigits(_) => "".into(),
         FacetType::Length(value) => gen_length_validation(value.as_str(), name).into(),
-        FacetType::MaxExclusive(value) => gen_max_exclusive_validation(value.as_str(), name).into(),
-        FacetType::MaxInclusive(value) => gen_max_inclusive_validation(value.as_str(), name).into(),
+        FacetType::MaxExclusive(value) => {
+            gen_max_exclusive_validation(value.as_str(), name, typename).into()
+        }
+        FacetType::MaxInclusive(value) => {
+            gen_max_inclusive_validation(value.as_str(), name, typename).into()
+        }
         FacetType::MaxLength(value) => gen_max_length_validation(value.as_str(), name).into(),
-        FacetType::MinExclusive(value) => gen_min_exclusive_validation(value.as_str(), name).into(),
-        FacetType::MinInclusive(value) => gen_min_inclusive_validation(value.as_str(), name).into(),
+        FacetType::MinExclusive(value) => {
+            gen_min_exclusive_validation(value.as_str(), name, typename).into()
+        }
+        FacetType::MinInclusive(value) => {
+            gen_min_inclusive_validation(value.as_str(), name, typename).into()
+        }
         FacetType::MinLength(value) => gen_min_length_validation(value.as_str(), name).into(),
         _ => "".into(), // TODO: All Facet Types
     }
 }
 
-fn gen_max_exclusive_validation(value: &str, name: &str) -> String {
+fn gen_max_exclusive_validation(value: &str, name: &str, typename: &str) -> String {
     format!(
         r#"
-        if self.{name} >= "{value}".parse().unwrap() {{
+        if self.{name} >= "{value}".parse::<{typename}>().unwrap() {{
             return Err(format!("MaxExclusive validation error: invalid value of {name}! \nExpected: {name} < {value}.\nActual: {name} == {{}}", self.{name}));
         }}"#,
         name = name,
-        value = value
+        value = value,
+        typename = typename
     )
 }
 
-fn gen_max_inclusive_validation(value: &str, name: &str) -> String {
+fn gen_max_inclusive_validation(value: &str, name: &str, typename: &str) -> String {
     format!(
         r#"
-        if self.{name} > "{value}".parse().unwrap() {{
+        if self.{name} > "{value}".parse::<{typename}>().unwrap() {{
             return Err(format!("MaxInclusive validation error: invalid value of {name}! \nExpected: {name} <= {value}.\nActual: {name} == {{}}", self.{name}));
         }}"#,
         name = name,
-        value = value
+        value = value,
+        typename = typename
     )
 }
 
 fn gen_length_validation(value: &str, name: &str) -> String {
+    let value: u32 = value.parse().unwrap();
     format!(
         r#"
-        if self.{name}.len() != "{value}".parse().unwrap() {{
+        if self.{name}.len() != {value} {{
             return Err(format!("Length validation error. \nExpected: {name} length == {value} \nActual: {name} length == {{}}", self.{name}.len()));
         }}"#,
         name = name,
@@ -75,9 +86,10 @@ fn gen_length_validation(value: &str, name: &str) -> String {
 }
 
 fn gen_max_length_validation(value: &str, name: &str) -> String {
+    let value: u32 = value.parse().unwrap();
     format!(
         r#"
-        if self.{name}.len() > "{value}".parse().unwrap() {{
+        if self.{name}.len() > {value} {{
             return Err(format!("MaxLength validation error. \nExpected: {name} length <= {value} \nActual: {name} length == {{}}", self.{name}.len()));
         }}"#,
         name = name,
@@ -85,32 +97,39 @@ fn gen_max_length_validation(value: &str, name: &str) -> String {
     )
 }
 
-fn gen_min_exclusive_validation(value: &str, name: &str) -> String {
+fn gen_min_exclusive_validation(value: &str, name: &str, typename: &str) -> String {
     format!(
         r#"
-        if self.{name} <= "{value}".parse().unwrap() {{
+        if self.{name} <= "{value}".parse::<{typename}>().unwrap() {{
             return Err(format!("MinExclusive validation error: invalid value of {name}! \nExpected: {name} > {value}.\nActual: {name} == {{}}", self.{name}));
         }}"#,
         name = name,
-        value = value
+        value = value,
+        typename = typename
     )
 }
 
-fn gen_min_inclusive_validation(value: &str, name: &str) -> String {
+fn gen_min_inclusive_validation(value: &str, name: &str, typename: &str) -> String {
     format!(
         r#"
-        if self.{name} < "{value}".parse().unwrap() {{
+        if self.{name} < "{value}".parse::<{typename}>().unwrap() {{
             return Err(format!("MinInclusive validation error: invalid value of {name}! \nExpected: {name} >= {value}.\nActual: {name} == {{}}", self.{name}));
         }}"#,
         name = name,
-        value = value
+        value = value,
+        typename = typename
     )
 }
 
 fn gen_min_length_validation(value: &str, name: &str) -> String {
+    let value: u32 = value.parse().unwrap();
+    if value == 0 {
+        return "".into();
+    }
+
     format!(
         r#"
-        if self.{name}.len() < "{value}".parse().unwrap() {{
+        if self.{name}.len() < {value} {{
             return Err(format!("MinLength validation error. \nExpected: {name} length >= {value} \nActual: {name} length == {{}}", self.{name}.len()));
         }}"#,
         name = name,
@@ -135,25 +154,25 @@ mod test {
     #[test]
     fn test_gen_max_exclusive_validation() {
         let expected = r#"
-        if self.count >= "5".parse().unwrap() {
+        if self.count >= "5".parse::<i32>().unwrap() {
             return Err(format!("MaxExclusive validation error: invalid value of count! \nExpected: count < 5.\nActual: count == {}", self.count));
         }"#;
-        assert_eq!(gen_max_exclusive_validation("5", "count"), expected);
+        assert_eq!(gen_max_exclusive_validation("5", "count", "i32"), expected);
     }
 
     #[test]
     fn test_gen_max_inclusive_validation() {
         let expected = r#"
-        if self.count > "5".parse().unwrap() {
+        if self.count > "5".parse::<i32>().unwrap() {
             return Err(format!("MaxInclusive validation error: invalid value of count! \nExpected: count <= 5.\nActual: count == {}", self.count));
         }"#;
-        assert_eq!(gen_max_inclusive_validation("5", "count"), expected);
+        assert_eq!(gen_max_inclusive_validation("5", "count", "i32"), expected);
     }
 
     #[test]
     fn test_gen_length_validation() {
         let expected = r#"
-        if self.name.len() != "50".parse().unwrap() {
+        if self.name.len() != 50 {
             return Err(format!("Length validation error. \nExpected: name length == 50 \nActual: name length == {}", self.name.len()));
         }"#;
         assert_eq!(gen_length_validation("50", "name"), expected);
@@ -162,36 +181,42 @@ mod test {
     #[test]
     fn test_gen_max_length_validation() {
         let expected = r#"
-        if self.name.len() > "50".parse().unwrap() {
+        if self.name.len() > 50 {
             return Err(format!("MaxLength validation error. \nExpected: name length <= 50 \nActual: name length == {}", self.name.len()));
         }"#;
-        assert_eq!(gen_max_length_validation("50", "name"), expected);
+        assert_eq!(gen_max_length_validation("50", "name",), expected);
     }
 
     #[test]
     fn test_gen_min_exclusive_validation() {
         let expected = r#"
-        if self.count <= "5".parse().unwrap() {
+        if self.count <= "5".parse::<i32>().unwrap() {
             return Err(format!("MinExclusive validation error: invalid value of count! \nExpected: count > 5.\nActual: count == {}", self.count));
         }"#;
-        assert_eq!(gen_min_exclusive_validation("5", "count"), expected);
+        assert_eq!(gen_min_exclusive_validation("5", "count", "i32"), expected);
     }
 
     #[test]
     fn test_gen_min_inclusive_validation() {
         let expected = r#"
-        if self.count < "5".parse().unwrap() {
+        if self.count < "5".parse::<i32>().unwrap() {
             return Err(format!("MinInclusive validation error: invalid value of count! \nExpected: count >= 5.\nActual: count == {}", self.count));
         }"#;
-        assert_eq!(gen_min_inclusive_validation("5", "count"), expected);
+        assert_eq!(gen_min_inclusive_validation("5", "count", "i32"), expected);
     }
 
     #[test]
     fn test_gen_min_length_validation() {
         let expected = r#"
-        if self.name.len() < "50".parse().unwrap() {
+        if self.name.len() < 50 {
             return Err(format!("MinLength validation error. \nExpected: name length >= 50 \nActual: name length == {}", self.name.len()));
         }"#;
         assert_eq!(gen_min_length_validation("50", "name"), expected);
+    }
+
+    #[test]
+    fn test_gen_min_length_zero_validation() {
+        let expected = "";
+        assert_eq!(gen_min_length_validation("0", "name"), expected);
     }
 }

